src/main/java/net/imglib2/cache/util/FunctorLoader.java

```diff
@@ -0,0 +1,44 @@
+package net.imglib2.cache.util;
+
+import java.util.stream.IntStream;
+
+import net.imglib2.FinalInterval;
+import net.imglib2.Interval;
+import net.imglib2.cache.CacheLoader;
+import net.imglib2.img.cell.Cell;
+import net.imglib2.img.cell.CellGrid;
+
+public class FunctorLoader< A > implements CacheLoader< Long, Cell< A > >
+{
+
+	public interface Functor< T, R > {
```

Comment:
Thanks for the clarification @tpietzsch 
For the use case of this pull request -- because it is specific to non-throwing `CellImg` only -- I now think that we should not allow checked exceptions in the `IntervalKeyLoaderAsLongKeyLoader` to follow suit. This could either be with a `NonThrowingCacheLoader` as mentioned above or with a `Function` as @dietzc and @ctrueden suggested which would save as an additional interface (preferable).